agent/consul/catalog_endpoint.go

```diff
@@ -274,7 +274,11 @@ func (c *Catalog) ServiceNodes(args *structs.ServiceSpecificRequest, reply *stru
 			}
 
 			if args.TagFilter {
-				return s.ServiceTagNodes(ws, args.ServiceName, args.ServiceTags)
+				tags := args.ServiceTags
+				if len(args.ServiceTag) > 0 {
+					tags = append(tags, args.ServiceTag)
+				}
+				return s.ServiceTagNodes(ws, args.ServiceName, tags)
```

Comment:
This seems like it will merge old and new fields...

Assuming that the old field is only set by <1.3 clients who didn't know about the new one I think it will do the job, but that assumption isn't mentioned in comments so I think it will get lost over time!

My marginal preference would be to make the actual logic explicit without that assumption:

```go
if args.TagFilter {
  tags := args.ServiceTags
  // If there are no tags, fallback to the old singular tag field as this may be an RPC from a client running Consul < 1.3.0
  if len(tags) == 0 && len(args.ServiceTag) > 0 {
    tags = []string{args.ServiceTag} // Forget the actual type so it might not be this...
  }
  return s.ServiceTagNodes(ws, args.ServiceName, tags)
}
```